R/merge.R

```diff
@@ -49,9 +55,14 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
     if (!length(by))
       by = intersect(nm_x, nm_y)
     if (length(by) == 0L || !is.character(by))
-      stopf("A non-empty vector of column names for `by` is required.")
-    if (!all(by %chin% intersect(nm_x, nm_y)))
-      stopf("Elements listed in `by` must be valid column names in x and y")
+      stopf("A non-empty vector of column names for by is required.")
+    missing_in_x <- setdiff(by, nm_x)
+    missing_in_y <- setdiff(by, nm_y)
+    if (length(missing_in_x) > 0 || length(missing_in_y) > 0) {
+      stopf("The following columns are missing:%s%s",
+            if (length(missing_in_x) > 0) sprintf(" - From x: %s", toString(missing_in_x)) else "",
```

Comment:
Use `gettextf()` to enable translation

Reply:
To clarify, I meant to swap `sprintf()` for `gettextf()`. `stopf()` is already equipped to run `gettext()`:

https://github.com/Rdatatable/data.table/blob/2714cb43262bc4dc2c25810030175db8727964f1/R/translation.R#L20-L22